### PR TITLE
Add a handler to API error when submitting the ticket on the Migrate Message step

### PR DESCRIPTION
**Pull request by valterlorran**
Related to #

## Proposed Changes

* We noticed that we were not handling the errors when trying to submit the ticket via the Migrate Message step.
* I added a simple message when the request fails.
* We also didn't have track events for success or failure, I added the following events:
  *  calypso_importer_migration_ticket_submit_success
  * calypso_importer_migration_ticket_submit_error
* I also had to update the revamp flow to reflect the change we made.

Captura de Tela 2024-09-09 às 23 05 53

In a follow-up PR, I will also update the flow for the import-hosted-site flow.

## Testing Instructions

First we need to make the ticket creation endpoint return an error, so we can see the error message.
* Make sure you are sandboxed
* Edit the method below to always return an error:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Serfg%2Qncv%2Qcyhtvaf%2Sraqcbvagf%2Suryc%2Qzvtengvba%2Qgvpxrg.cuc%3Se%3Q487565o5%2361-og
```
return new WP_Error('test','message');
```

* Now go through the migration flow by navigating to `/start`:
  * Go through the domain selection step
  * Select the free plan on the `Plans` page
  * On the `Goals` step, select the "Import existing content or website" option and click continue
  * Input the source site URL in the Identify step
  * Next, select the "Migrate Site" option on the `Import or Migrate` step
  * Select "Do it for me" on the `How to migrate` step
  * Go through the checkout
  * Once you complete the checkout, you should land on the new `Credentials` step
  * Now skip the step, and you should be redirected to the Migrate Message step and see the error message.
  * You can also check the track events are being fire correctly
  * You can also check the successful state, by removing the error you introduced.

### Testing the revamp flow
* Navigate to `/setup/migration` and go through the assisted flow
* Don't forget to make the endpoint linked above return an error.
* Once you reach the credentials step, skip it
* You should be redirected back to the Credentials step with an error message.
* You can also undo the error on the endpoint to ensure the ticket is created and you don't see an error.

**Note:** Don't forget to at least leave a message in your ticket information that it was a test and asking the HEs to close it.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by fditrapani** [comment]
Thanks for kicking this off @valterlorran. We should definitely be handling this error but we also need to give people a way to try again. There's currently no way to get back to the form. What do you think if we show something like this instead:

image

File: client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx [hunk 1]
```diff
@@ -38,7 +39,24 @@ const ImporterMigrateMessage: Step = () => {
 	const fromUrl = useQuery().get( 'from' ) || '';
 	const siteSlug = siteSlugParam ?? '';
 	const isCredentialsSkipped = useQuery().get( 'credentials' ) === 'skipped';
-	const { isPending, sendTicket } = useSubmitMigrationTicket();
+	const [ hasError, setHasError ] = useState< APIError | null >( null );
+	const { isPending, sendTicket } = useSubmitMigrationTicket( {
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx:41]:
```suggestion
	const { isPending, sendTicket, hasError, error } = useSubmitMigrationTicket( {
```

You can capture the error state directly from the migration.

File: client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx [hunk 2]
```diff
@@ -129,44 +147,53 @@ const ImporterMigrateMessage: Step = () => {
...
-										'You are all set! Our Happiness Engineers will be reaching out to you shortly at <strong>%(email)s</strong> to help you migrate <strong>%(webSite)s</strong> to WordPress.com.'
+					{ ! isPending && hasError && (
+						<div className="migration-message__request-error">
+							{ __( 'An error occurred while processing your request. Please try again later.' ) }
+						</div>
+					) }
+					{ ! isPending && ! hasError && (
+						<>
+							<div className="message">
+								{ createInterpolateElement(
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx]: It seems an outdated way to do translation with formatting. 

We have less verbose ways to do it, like this example:

https://github.com/Automattic/wp-calypso/blob/4440ddbc2b37b95cc56c947fd094e552373138b9/client/a8c-for-agencies/sections/marketplace/download-products-form/index.tsx#L89-L97

**Comment by gabrielcaires** [comment]
I just saw the @fditrapani comments after my initial comments. 
Yes, we need design support; the current approach doesn't permit the user to try again.

**Comment by fditrapani** [comment]
Let me know if what I proposed looks good or if you all need anything else!

File: client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx (hunk 2 above)
Comment (valterlorran) [reply client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx]: We should create a new issue for this change. I will do it shortly.

**Comment by valterlorran** [comment]
@fditrapani and @gabrielcaires, this error handling is for the "Migrate Message" step, in the case the ticket creation in that page fails. Knowing that, should we still redirect the user back to the Credentials step?

File: client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx (hunk 2 above)
Comment (valterlorran) [reply client/landing/stepper/declarative-flow/internals/steps-repository/importer-migrate-message/index.tsx]: Done: https://github.com/Automattic/wp-calypso/issues/94258

**Comment by fditrapani** [comment]
@valterlorran can you clarify which step the Migrate Message step is?

**Comment by valterlorran** [comment]
@fditrapani, I mean the step that shows the instructions for the DIFM flow. In case the user skips the Credentials step, the ticket will be created on the instruction step(Migrate Message), so in case that ticket creation fails somehow, we need to display the error, and not the instructions.

**Comment by valterlorran** [comment]
We should already be handling that on the back end. I can look for ways to improve ticket creation and add mechanisms to alert us about it. However, this endpoint is not supposed to fail, but somehow, it happened a few times. We should handle the request on the front end for cases like connection loss or outage in our platform.

**Comment by fditrapani** [comment]
Thanks for the extra background. Makes sense. 

In that case, I think it makes most sense to return back to the form with an error that asks people to try again. This gives them something to action on. Hopefully, it'll work on a second or third attempt.

Perhaps, we might also want to keep people on the form while we do all our processing so that the transition isn't so jarring. I designed this state in the event that we needed time to verify credentials:

image

**Comment by valterlorran** [comment]
Yep, I agree with that! I will work on this on Monday.

**Comment by valterlorran** [comment]
Thanks for the feedback, @fditrapani; I applied your suggestions with a slight change in the component's position, as we discussed.

**Comment by fditrapani** [comment]
Thanks for taking care of that. I can't test this locally but it looks great from the screenshot!

**Comment by andres-blanco** [comment]
@valterlorran you'll need to rebase this PR since there was a pretty big refactor on the credentials form recently: https://github.com/Automattic/wp-calypso/pull/94409

**Comment by valterlorran** [comment]
@andres-blanco, done! It's ready for review again.

**Comment by andres-blanco** [comment]
@valterlorran same comment as in the other PR. Do we need to make adjustments to the revamped flow?

**Review by andres-blanco — APPROVED** [review]
This is working as expected.

I just don't like the small font on the error message, but probably that's me being old and can be dealt in a follow-up. @fditrapani, this font size is expected to you?
image

**Comment by fditrapani** [comment]
Thanks for the ping @andres-blanco. It does look small in relation to the rest of the ui. This is what we get from the default component we're using. I don't think it'll be a good idea to update the component at this time.

We could perhaps try passing the copy to the `title` vs the `description` attribute? I'll bold the copy and perhaps make it stand apart from the rest of the page a little better.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16781127).

**Pull request merged**